roomserver/internal/input/input_events.go

```diff
@@ -67,14 +68,15 @@ var processRoomEventDuration = prometheus.NewHistogramVec(
 // nolint:gocyclo
 func (r *Inputer) processRoomEvent(
 	ctx context.Context,
+	updater *shared.RoomUpdater,
 	input *api.InputRoomEvent,
-) (err error) {
+) (commit bool, err error) {
```

Comment:
I would really prefer this to be a dedicated type and not a bool. It's unclear as it stands because we also have functions which return `bool, error` but mean very different things: `retry, err := r.processRoomEventUsingUpdater(context.Background(), roomID, &inputRoomEvent)` something like:
```
type Retry bool
var RetryEvent Retry = true
var DoNotRetry Retry = false
```
then use `Retry` instead of `bool`. Ditto for the commit stuff. It makes it a lot clearer to see `Commit, nil` or `Rollback, err`, etc.
```